tests/manage/pv_services/test_delete_create_pvc_same_name.py

```diff
@@ -4,111 +4,91 @@
-    """
-    self = request.node.cls
-    self.interface_type = request.param
-
-    setup(self)
-    yield
-    teardown(self)
-
-
-def setup(self):
+@pytest.fixture()
+def resources(request):
     """
-    Creates the resources needed for the type of interface to be used.
+    Delete the pvc resources and validate pv deletion created during the test
 
-    For CephBlockPool interface: Creates CephBlockPool, Secret and StorageClass
-    For CephFilesystem interface: Creates Secret and StorageClass
+    Returns:
+        list: empty list of pvcs
     """
-    logger.info(f"Creating resources for {self.interface_type} interface")
-
-    self.interface_name = None
-    if self.interface_type == constants.CEPHBLOCKPOOL:
-        self.cbp_obj = helpers.create_ceph_block_pool()
-        assert self.cbp_obj, f"Failed to create block pool"
-        self.interface_name = self.cbp_obj.name
+    pvcs = []
 
-    elif self.interface_type == constants.CEPHFILESYSTEM:
-        self.interface_name = helpers.get_cephfs_data_pool_name()
+    def finalizer():
+        for instance in pvcs:
+            if not instance.is_deleted:
+                instance.delete()
+                instance.ocp.wait_for_delete(
+                    instance.name
+                )
+            assert helpers.validate_pv_delete(instance.backed_pv)
 
-    self.secret_obj = helpers.create_secret(interface_type=self.interface_type)
-    assert self.secret_obj, f"Failed to create secret"
-
-    self.sc_obj = helpers.create_storage_class(
-        interface_type=self.interface_type,
-        interface_name=self.interface_name,
-        secret_name=self.secret_obj.name
-    )
-    assert self.sc_obj, f"Failed to create storage class"
-
-
-def teardown(self):
-    """
-    Deletes the resources for the type of interface used.
-    """
-    logger.info(f"Deleting resources for {self.interface_type} interface")
-
-    resources_list = [PVC_OBJ, self.sc_obj, self.secret_obj]
-    if self.interface_type == constants.CEPHBLOCKPOOL:
-        resources_list.append(self.cbp_obj)
-
-    for resource in resources_list:
-        try:
-            logger.info(f"Deleting {resource.kind} {resource.name}")
-            resource.delete()
-        except AttributeError:
-            continue
-        except exceptions.CommandFailed:
-            logger.error(f"Deletion of {resource.kind} {resource.name} failed")
+    request.addfinalizer(finalizer)
+    return pvcs
 
 
 @tier2
 @pytest.mark.polarion_id("OCS-324")
 class TestDeleteCreatePVCSameName(ManageTest):
     """
-    Delete PVC and create a new PVC with same name
+    Automates the testcase OCS-324
+    OCS-324 - FT-OCP-PVCDeleteAndCreate-SameName: Delete PVC and create a new
+    PVC with same name
     """
-    def test_pvc_delete_create_same_name(self, test_fixture):
+    @pytest.mark.parametrize(
+        argnames="interface",
+        argvalues=[
+            pytest.param(*[constants.CEPHBLOCKPOOL]),
```

Comment:
should we separate to 2 TCs - FS and RBD?

Reply:
ack, I will create one more test case in polarion

Reply:
why to separate?

Reply:
@ebenahar By "separate to 2 TCs", did you meant in 2 TC in Polarion or in the script?

In the script, we don't need  separation.

Reply:
@sidhant-agrawal @RazTamir meant in Polarion

Reply:
created a test in polarion and updated the same in the script.

Reply:
@sidhant-agrawal thanks! great PR